libcaf_core/test/serial_reply.cpp

```diff
@@ -29,48 +29,50 @@ CAF_TEST(test_serial_reply) {
   actor_system system{cfg};
   auto mirror_behavior = [=](event_based_actor* self) -> behavior {
     self->set_default_handler(reflect);
-    return {[] {
-      // nop
-    }};
+    return {
+      [] {
+        // nop
+      },
+    };
   };
-  auto master = system.spawn([=](event_based_actor* self) {
+  auto master = system.spawn([=](event_based_actor* self) -> behavior {
     CAF_MESSAGE("ID of master: " << self->id());
     // spawn 5 mirror actors
     auto c0 = self->spawn<linked>(mirror_behavior);
     auto c1 = self->spawn<linked>(mirror_behavior);
     auto c2 = self->spawn<linked>(mirror_behavior);
     auto c3 = self->spawn<linked>(mirror_behavior);
     auto c4 = self->spawn<linked>(mirror_behavior);
-    self->become([=](int) mutable {
-      auto rp = self->make_response_promise();
-      CAF_MESSAGE("received 'hi there'");
-      self->request(c0, infinite, sub0_atom::value)
-        .then([=](sub0_atom) mutable {
+    return {
+      [=](hi_atom) mutable {
+        auto rp = self->make_response_promise();
+        CAF_MESSAGE("received 'hi there'");
+        self->request(c0, infinite, sub0_atom_v).then([=](sub0_atom) mutable {
```

Comment:
Just for my understanding: Why do these handlers have to be marked `mutable`? Wouldn't referencing the context be sufficient?

Reply:
We call `deliver` on the captured response promise, which requires non-const access.